core/drivers/rstctrl/stm32_rstctrl.h

```diff
@@ -0,0 +1,34 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2018-2023, STMicroelectronics
+ */
+
+#include <drivers/rstctrl.h>
+
+/* Exposed rstctrl instance */
+struct stm32_rstline {
+	unsigned int id;
+	struct rstctrl rstctrl;
+	const struct stm32_reset_data *data;
+	SLIST_ENTRY(stm32_rstline)link;
+};
+
+struct stm32_rstline *to_rstline(struct rstctrl *rstctrl);
+
+struct stm32_reset_cfg {
+	uint16_t offset;
```

Comment:
These two can be unsigned int. No need to restrain IMO, operations will be faster